var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -40,207 +130,234 @@ class FoamExtend(Package):
-    depends_on('metis@5:', when='+metis')
-    depends_on('parmetis', when='+parmetis')
-    depends_on('parmgridgen', when='+parmgridgen')
+    depends_on('scotch~metis',     when='~ptscotch+scotch')
+    depends_on('scotch~metis+mpi', when='+ptscotch')
+    depends_on('metis@5:',         when='+metis')
+    depends_on('parmetis',         when='+parmetis')
+    depends_on('parmgridgen',      when='+parmgridgen')
+    depends_on('paraview@:5.0.1',  when='+paraview')
+
+    # Some user settings, to be adjusted manually or via variants
+    foam_cfg = {
+        'WM_COMPILER':        'Gcc',  # <- %compiler
+        'WM_ARCH_OPTION':     '64',   # (32/64-bit on x86_64)
+        # FUTURE? 'WM_LABEL_SIZE':      '32',  # <- +int64
+        'WM_PRECISION_OPTION': 'DP',  # <- +float32
+        'WM_COMPILE_OPTION':  'SPACKOpt',  # Do not change
+        'WM_MPLIB':           'SPACK',     # Do not change
+    }
+
+    # The system description is frequently needed
+    foam_sys = {
+        'WM_ARCH':      None,
+        'WM_COMPILER':  None,
+        'WM_OPTIONS':   None,
+    }
+
+    build_script = './spack-Allwmake'  # <- Generated by patch() method.
+    # phases = ['configure', 'build', 'install']
 
-    depends_on('paraview', when='+paraview')
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('FOAM_INST_DIR', self.prefix)
+        run_env.set('WM_PROJECT_DIR', self.projectdir)
+
+    @property
+    def _canonical(self):
+        """Canonical name for this package and version"""
+        return 'foam-extend-{0}'.format(self.version.up_to(2))
+
+    @property
+    def projectdir(self):
+        """Absolute location of project directory: WM_PROJECT_DIR/"""
+        return join_path(self.prefix, self._canonical)  # <- prefix/canonical
+
+    @property
+    def etc(self):
+        """Absolute location of the OpenFOAM etc/ directory"""
+        return join_path(self.projectdir, 'etc')
+
+    @property
+    def archbin(self):
+        """Relative location of architecture-specific executables"""
+        wm_options = self.set_openfoam()
+        return join_path('applications', 'bin', wm_options)
+
+    @property
+    def archlib(self):
+        """Relative location of architecture-specific libraries"""
+        wm_options = self.set_openfoam()
+        return join_path('lib', wm_options)
+
+    @property
+    def mplib_content(self):
+        """Define 'SPACK' mpi settings to have wmake
+        use spack information with minimum modifications to OpenFOAM
+        """
+        return [
+            'PFLAGS = -DOMPI_SKIP_MPICXX -DMPICH_IGNORE_CXX_SEEK',
+            'PINC   = -I{0}'.format(self.spec['mpi'].prefix.include),
+            'PLIBS  = -L{0} -lmpi'.format(pkglib(self.spec['mpi']))
+        ]
+
+    @property
+    def rpath_info(self):
+        """Define 'SPACKOpt' compiler optimization file to have wmake
+        use spack information with minimum modifications to OpenFOAM
```

Comment:
@olesenm: I looked a little harder at the build.  It looks like you do this:

```python
    wm_compiler = self.supported_compilers[self.compiler.name]
```

and then you use this to pick an existing configuration file in OpenFOAM's build.  

Suppose you picked the `Gcc` configuration.  If that configuration ends up just trying to execute `gcc` and `g++`, then you shouldn't need to set `RPATH`s at all -- you'll pick up our wrappers (named `gcc` and `g++` for GCC builds, `icc` and `icpc` for Intel builds, etc.), which already add `-I`, `-L`, and `-Wl,-rpath` flags for dependency include and lib(64) directories.  We also set `CC`, `CXX`, and friends to point at the wrappers, so if your build automatically picks up the compiler from those, then it should work fine without manually setting any `RPATH`s.

If, on the other hand, `wmake` has its own fancy compiler search and tries to find an absolute path to some compiler, you may end up circumventing Spack's wrappers.  If you look at the `lua`,  `parpack`, or `wannier90` builds (e.g., try `spack edit wannier90`) -- they all manually substitute the Spack compilers into a build in one way or another.  Typically that is necessary for plain `Makefile` builds.  You may need to do this if the build doesn't call plain `gcc` or if it does not respect `CC`, `CXX`, `F77`, and `FC` environment variables.

Does that make sense?